### ISSUE TITLE
checker.py crashes if the ServiceProviderConfig endpoint returns a 401 status code

### ISSUE THREAD

**Issue by fflorent** [yaal-coop/scim2-tester#3]
Hello :wave: 

## Context

I ran the following command:

```bash
$ poetry run python ./scim2_tester/checker.py http://MY_SERVER/scim
```

I got the following error:
```bash
Traceback (most recent call last):
  File "/home/florentpro/projects/scim2-tester/./scim2_tester/checker.py", line 104, in <module>
    results = check_server(scim)
              ^^^^^^^^^^^^^^^^^^
  File "/home/florentpro/projects/scim2-tester/./scim2_tester/checker.py", line 82, in check_server
    for resource_type in resource_types:
TypeError: 'Response' object is not iterable
```

## Expected behavior

I would have expected a proper error displaying what is wrong with my server, which was that the `/ServiceProviderConfig` returned a 401 status code with no authentication.

I wonder if the server is expected to return the result when not authenticated or can reject the request.

If I have time, I can take a look and propose a fix, but currently I am a bit in a rush

**Comment by azmeuk** [comment]
Thank you for your report. Indeed, the project is still young and most error case are not covered at the moment. It should indeed correctly handle unexpected error statuses.

**Comment by azmeuk** [comment]
I just released 0.1.4 which fixes this. Can you confirm me it behaves the way you expect it?

**Comment by fflorent** [comment]
It indeed solves the issue, thanks!

I have other surprising problems, unrelated to the issue here, and was present before the update. You can close this issue, I may bother you in another one :)
